### DISCUSSION TITLE
CLI support for a modular file structure

### DISCUSSION THREAD

**Discussion by Gim3l** [blitz-js/blitz#1434] [Ideas]
To offer a slighly better UX for developing apps with Blitz I have a proposal. In blitz I find it customary to keep my file structure clean by creating all _modules_ in a modules directory. For example, if a segment of my app deals with projects I tend to place all files related to projects (pages, components, queries, mutations, hooks) in a `modules > projects` directory. 

It would be nice if the CLI supports something like `blitz generate module projects` which would create a sub folder in a `modules` directory. The generated folders in `modules > projects` could be a queries, components, pages and mutations directories. Then we could have the CLI allow us to use a module with  the `blitz module use projects` command
which could allow us to run the generator for the current module with `bliz module generate [component|page|etc..]`. Would the cli be capable of doing something like that? Is this a good idea? I'm drawing inspiration from the way the laravel modules package works.

For reference:
https://nwidart.com/laravel-modules/v1/advanced-tools/artisan-commands

**Comment by theZieger** [comment]
The one and only thing I really liked about other frameworks (ember, angular) was the superior CLI and code scaffolding features (since they are opinionated frameworks). So :+1: on more code scaffolding commands for the CLI tool.

See: https://cli.emberjs.com/release/basic-use/cli-commands/#generatemorefiles
or: https://angular.io/cli/generate

Anyway...

Maybe I'm on the wrong track here but isn't Blitz already creating a file/folder structure in the way you described?
The only difference would be that it's not `modules/projects` but `app/projects`. Within the projects folder you should find related queries, pages, components and so on (if you generated the `projects` model via blitz `generate`).

So the only difference I see in your question is that you want the CLI to have more broken down commands (which I think would be great to have :slightly_smiling_face: )?

**Comment by flybayer** [comment]
Fyi, here's a related discussion: https://github.com/blitz-js/blitz/discussions/1183
